File: python/src/Behavioral/Visitor/TitleLongBlurbVisitor.py
#//TitleLongBlurbVisitor - one of two concrete Visitors

from TitleBlurbVisitor import TitleBlurbVisitor
from BookInfo import BookInfo
from DvdInfo import DvdInfo
from GameInfo import GameInfo

class TitleLongBlurbVisitor(TitleBlurbVisitor):

	def visit (self, info):
		if isinstance(info,BookInfo):
			self.setTitleBlurb("LB-Book: " + info.titleName + ", Author: " + info.author)
		elif isinstance(info,DvdInfo):
			self.setTitleBlurb("LB-DVD: " + info.titleName + ", starring " + info.star + ", encoding region: " + info.region)
		elif isinstance(info,GameInfo):
			self.setTitleBlurb("LB-Game: " + info.titleName )

File: ruby/src/Behavioral/Visitor/TitleLongBlurbVisitor.rb
#//TitleLongBlurbVisitor - one of two concrete Visitors

#require "Behavioral/Visitor/TitleBlurbVisitor"
#require "Behavioral/Visitor/BookInfo"
#require "Behavioral/Visitor/DvdInfo"
#require "Behavioral/Visitor/GameInfo"

class TitleLongBlurbVisitor < TitleBlurbVisitor

	def visit ( info)
		if info.is_a? BookInfo
			setTitleBlurb("LB-Book: #{info.titleName}, Author: #{info.author}")
		elsif info.is_a? DvdInfo
			setTitleBlurb("LB-DVD: #{ info.titleName}, starring #{info.star }, region: #{ info.region }")
		elsif info.is_a? GameInfo
			setTitleBlurb("LB-Game: #{info.titleName }")
		end
	end
end
